### DISCUSSION TITLE
Extend API to accept 3phase voltage, used kWh etc?

### DISCUSSION THREAD

**Discussion by stevoh6** [serkri/SmartEVSE-3#117] [Ideas]
I have esp as smart metter over wifi, I would like to pass all needed values ( as modbus smart metters do), (I dont know which all values are need) over the API call. It is possible? Whats needs to be changed to be able use wireless "dyi esp smart" meter? I can help develop it, if you push me to right drection :-)

**Comment by dingo35** [comment]
Sorry stevo, your requirements are so specific that is doesnt make sense to put this in the mainline firmware. The choice for modbus as primary communications method with high reliability makes sense, you wouldnt want your neighbour interfering with your wifi and blow up your mains fuses...


Again feel free to adapt the firmware to your own requirements..

**Reply by stevoh6** [reply]
No problem at all :) But be able to see kwh and other stuff (as they can be seen when modbus is present) will  be nice. Could you please atleast help me understand which values are most important? Only voltage and current? Or also power factor?

Actually I make two topics regarding this, because they are two separate things.

**Comment by dingo35** [comment]
Most important are currents, voltages and power factor are not used in the firmware.

The energy (kWh) and power (kW) that are displayed are directly taken off of the modbus.
